pages/SetPasswordPage.js

```diff
@@ -0,0 +1,161 @@
+import React, { useState } from 'react';
+import { StyleSheet, View, Linking, Alert } from 'react-native';
+import {Button, Text } from 'galio-framework';
+import { CustomInputPassword } from '../components/CustomInput';
+import { validateSetPassword, showAlertForInvalidInput } from '../components/Validation';
+import { setToken, getToken } from '../components/TokenManager';
+import qs from 'qs';
+
+const SetPasswordPage = (props) => {
+
+  const { navigation } = props;
+  const [ user, setUser] = useState({});
+  const [ errors, setErrors] = useState({});
+
+  const handleInput = (value, name) => {
+      setUser({ ...user, [name]: value})
+      console.log(user)
+  }
+
+  const extractTokenFromURL = () => {
+    console.log("In extract token")
+    Linking.getInitialURL().then((url) => {
+      if (url) {
+        console.log('Initial url is: ' + url); 
+
+        //url1 = "app://boxapp/setpassword?confirm=NPKZ9qxgWypV7Qmei1eP"
+        let urlArray=url.split("?")
+        let queryParams = urlArray[1]
+        let setPasswordToken = qs.parse(queryParams).confirm
+        console.log(setPasswordToken)
+        setToken('setPasswordToken', setPasswordToken)
+        return setPasswordToken
+      }
+    }).catch(err => console.error(err));
+  } 
+
+
+  const setPasswordForUser = () => {
+    console.log("In set password for user method")
+    
+    fetch("http://192.168.1.69:3000/api/v1/setpwd",
+    {
+      method: 'POST',
+      headers:{
+        "Content-Type":"application/json"
+      },
+      body: JSON.stringify({
+        user: {
+          confirmation_token: extractTokenFromURL(),
+          password: user.password,
+          password_confirmation: user.confirmPassword,
+        }      
+      })
+    }).then((result) => {
+      if(result.status === 200){
```

Comment:
Backend will return 201 status code in this case.
instead of checking for 200, check for range between 200 to 299

Also, you can use result.ok instead of result.status === 200